### PR TITLE
[IDEA Plugin] - Add ballerina flavored markdown syntax support 

### PR DESCRIPTION
**Pull request by NipunaRanasinghe**
## Purpose
Update the documentation syntax to new ballerina flavored markdown syntax

## Samples
screenshot from 2018-07-31 14-50-12

### CODE REVIEW

File: tool-plugins/intellij/gradle.properties [hunk 1]
```diff
@@ -17,5 +17,5 @@
 ballerinaPluginName = ballerina-intellij-idea-plugin
 ballerinaPluginVersion = 0.980.0
 
-ideaVersion = 2016.3
+ideaVersion = 2017.3
```
Comment (Shan1024) [inline tool-plugins/intellij/gradle.properties]: We don't need to change the version, right?

File: tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/formatter/BallerinaBlock.java [hunk 2]
```diff
@@ -133,8 +132,8 @@ public Alignment getAlignment() {
             Alignment alignment = getAlignment(child);
             Indent indent = calculateIndent(child);
             Wrap wrap = createWrap(child);
-            blocks.add(new BallerinaBlock(child, alignment, indent, wrap, mySettings, mySpacingBuilder,
-                    myAlignmentMap));
+            blocks.add(
+                    new BallerinaBlock(child, alignment, indent, wrap, mySettings, mySpacingBuilder, myAlignmentMap));
```
Comment (Shan1024) [inline tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/formatter/BallerinaBlock.java]: Improve the formatting here.

File: tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/grammar/Ballerina.bnf [hunk 3]
```diff
@@ -200,15 +200,16 @@ Definition ::= ( TypeDefinition
     recoverWhile=TopLevelDefinitionRecover
 }
 
-private TopLevelDefinitionRecover ::= !(DOCUMENTATION_TEMPLATE_START|DEPRECATED_TEMPLATE_START|'@'|native|public|type|typedesc|service|function|enum|annotation|endpoint|record|int|float|boolean|string|byte|map|xml|xmlns|json|table|any|stream|object|future|identifier|'{')
+private TopLevelDefinitionRecover ::= !(DOCUMENTATION_TEMPLATE_START|MARKDOWN_DOCUMENTATION_LINE_START|PARAMETER_DOCUMENTATION_START|RETURN_PARAMETER_DOCUMENTATION_START|DEPRECATED_TEMPLATE_START|'@'|native
+|public|type|typedesc|service|function|enum|annotation|endpoint|int|float|boolean|string|byte|map|xml|xmlns|json|table|any|stream|object|future|identifier|'{')
```
Comment (Shan1024) [inline tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/grammar/Ballerina.bnf]: Can we merge these two lines?

File: tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/grammar/Ballerina.bnf [hunk 4]
```diff
@@ -954,3 +955,44 @@ singleBackTickDocInlineCode ::= SB_DOC_INLINE_CODE_START SINGLE_BACK_TICK_INLINE
 doubleBackTickDocInlineCode ::= DB_DOC_INLINE_CODE_START DOUBLE_BACK_TICK_INLINE_CODE? DOUBLE_BACK_TICK_INLINE_CODE_END {pin=1}
 
 tripleBackTickDocInlineCode ::= TB_DOC_INLINE_CODE_START TRIPLE_BACK_TICK_INLINE_CODE? TRIPLE_BACK_TICK_INLINE_CODE_END {pin=1}
+
+
```
Comment (Shan1024) [inline tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/grammar/Ballerina.bnf]: Lets remove these redundant newlines.

File: tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/grammar/Ballerina.flex [hunk 5]
```diff
@@ -247,6 +248,60 @@ XML_COMMENT_CHAR = [^{}>\\-] | {XML_ESCAPED_SEQUENCE}
...
+// DOUBLE_BACKTICKED_MARKDOWN
+DOUBLE_BACKTICK_CONTENT =   (([^`\n] | {BACKTICK} [^`])* [\n])? ({MARKDOWN_DOCUMENTATION_LINE_START} ([^`\n] |
+{BACKTICK} [^`])* [\n]?)+ | ([^`\n] | {BACKTICK} [^`])+
+
+DOUBLE_BACKTICK_MARKDOWN_START = {BACKTICK} {BACKTICK}
+DOUBLE_BACKTICK_MARKDOWN_END = {BACKTICK} {BACKTICK}
+
+// TRIPLE_BACKTICKED_MARKDOWN
+TRIPLE_BACKTICK_CONTENT = (([^`\n] | {BACKTICK} [^`] | {BACKTICK} {BACKTICK} [^`])* [\n])? ({MARKDOWN_DOCUMENTATION_LINE_START} ([^`\n] | {BACKTICK} [^`] | {BACKTICK} {BACKTICK} [^`])* [\n]?)+
+                          |   ([^`\n] | {BACKTICK} [^`] | {BACKTICK} {BACKTICK} [^`])+
```
Comment (Shan1024) [inline tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/grammar/Ballerina.flex]: Extra space after `|`. Also can we add clarification parenthesis for long and unclear conditions (everywhere)?

File: tool-plugins/intellij/gradle.properties (hunk 1 above)
Comment (NipunaRanasinghe) [reply tool-plugins/intellij/gradle.properties]: Yeah we don't, reverted the version change

File: tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/grammar/Ballerina.bnf (hunk 3 above)
Comment (NipunaRanasinghe) [reply tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/grammar/Ballerina.bnf]: Done

File: tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/grammar/Ballerina.bnf (hunk 4 above)
Comment (NipunaRanasinghe) [reply tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/grammar/Ballerina.bnf]: Removed

File: tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/grammar/Ballerina.flex (hunk 5 above)
Comment (NipunaRanasinghe) [reply tool-plugins/intellij/src/main/java/io/ballerina/plugins/idea/grammar/Ballerina.flex]: Removed the extra spaces

**Pull request merged**